haskell/actions.bzl

```diff
@@ -341,14 +343,41 @@ def link_haskell_bin(ctx, object_files):
 
   dummy_static_lib = _create_dummy_archive(ctx)
 
+  dep_info = gather_dep_info(ctx)
+
+  # On macOS, in order to simulate "relpath" behavior and make the binary load
+  # shared libaries from relative paths, we need to postprocess it with
+  # install_name_tool.  (This is what the Bazel-provided `cc_wrapper.sh` does
+  # for cc rules.)
+  # For details: https://blogs.oracle.com/dipol/entry/dynamic_libraries_rpath_and_mac
+  if not is_darwin(ctx):
+    compile_output = ctx.outputs.executable
+  else:
+    compile_output = ctx.actions.declare_file(ctx.outputs.executable.basename + ".temp")
+    ctx.actions.run_shell(
```

Comment:
We previously removed all uses of `run_shell` and almost all uses of `use_default_shell_env` (expect in toolchain implementation) to improve hermeticity. I'm not sure how I feel about this. It also uses hardcoded paths. Maybe for MacOS it's okayish? @mboes thoughts?

Reply:
The path to `install_name_tool` is already hard-coded in the compiler wrapper script that Bazel uses on macOS:
https://github.com/bazelbuild/bazel/blob/f98a7a2fedb3e714cef1038dcb85f83731150246/tools/cpp/osx_cc_wrapper.sh#L31

I'm not aware of any specific hermeticity issues other than the above, but let me know what you think.  This case is tricky because `install_name_tool` mutates the binary you give it, and we have an arbitrary number of times it wants to run.

Reply:
`run_shell` is fine. It's `use_default_shell_env` that causes hermiticity problems. But @judah isn't using that here, so fine by me!

Reply:
The absolute path to `install_name_tool` is a bit unfortunate, but if Bazel does it too, then I'm okay with it. I guess the reasoning could be that on macOS the path to `install_name_tool` is as well-known and unique as the path to `sh` in every UNIX, and its behaviour as quasi immutable over time.